packages/superset-ui-core/src/query/buildQueryObject.ts

```diff
@@ -42,8 +43,17 @@ export default function buildQueryObject<T extends QueryFormData>(
   const { metrics, columns, orderby } = extractQueryFields(residualFormData, queryFields);
 
   const extras = extractExtras(formData);
+  // collect all filters for conversion to simple filters/freeform clauses
+  const filterFormData: {
```

Comment:
2 questions:
1. Should this functionality to be in `extractExtras`?
2. Should we add here also `filters and adhoc_filters` from `custom_form_data and override_form_data`?

Reply:
1. `extractExtras` and `processFilters` probably need to be slightly refactored now that there's more stuff going on here, but since this code is slightly convoluted I prefer to keep these changes as small as possible for now. Let's do a proper refactor soon when we know the full scope of changes required by native and cross filters.
2. I don't think we should to allow overriding filters, as it could break charts, but happy to open up this discussion if there are known use cases for it. Also, `custom_form_data` probably shouldn't include `filters` and `adhoc_filters`, as those are already defined in `QueryFormData`  and `QueryObject`.

Reply:
About 2. For example what will be behavior for a case:
I have table with server pagination and cross filtering: 
1. User clicks `Page 2` we have in `custom_form_data` data about server pagination
2. We have button in dashboards like `resetDashboard` so we need reset it to `Page 1` or stuff like that